From 4460165c2e8c3e8097a0c6adb42b8b432c79a98a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 15 Oct 2020 17:26:57 +0300
Subject: [PATCH] [SW-12754] gaudi: security: close rest open gaps

There were several gaps left open between the relevant registers.
Finally close them.

Change-Id: I191d87390ea5b89d4ac591af68342508524440f6
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 8f3eed7..900d7fd 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -523,15 +523,27 @@
 
 static void preboot_secure_set_prot_bits_if_rtr(u32 base)
 {
-	u32 mask, prot_base = base;
+	u32 mask, prot_base = base, reg;
 
 	/* offsets: 0 - 0x400 */
 	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 8);
 
-	/* offsets: 0x400 - 0x500 - allowed access */
+	/* offsets: 0x400 - 0x480 - close 3 regs gap between open registers */
+	prot_base += SNGL_REG_PROT_SIZE * 8;
+	reg = mmSIF_RTR_0_LBW_RANGE_PROT_HIT_AW + sizeof(u32);
+	mask = PROT_BIT(reg) & PROT_BIT(reg + sizeof(u32)) &
+		PROT_BIT(reg + 2 * sizeof(u32));
+	regmap_writel(mask, PRIV_PROT_REG(prot_base));
+
+	/* offsets: 0x480 - 0x500 - close 3 regs gap between open registers */
+	prot_base += SNGL_REG_PROT_SIZE;
+	reg = mmSIF_RTR_0_LBW_RANGE_PROT_HIT_AR + sizeof(u32);
+	mask = PROT_BIT(reg) & PROT_BIT(reg + sizeof(u32)) &
+		PROT_BIT(reg + 2 * sizeof(u32));
+	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x500 - 0x580 */
-	prot_base += SNGL_REG_PROT_SIZE * 10;
+	prot_base += SNGL_REG_PROT_SIZE;
 	mask = ~(PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_8) &
 		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_9) &
 		 PROT_BIT(mmSIF_RTR_0_LBW_RANGE_PROT_MAX_AR_10) &
@@ -659,7 +671,6 @@
 
 static void preboot_secure_psoc_global_conf(void)
 {
-	int i;
 	u32 mask, prot_base, base = mmPSOC_GLOBAL_CONF_BASE & ~HL_CFG_BASE;
 
 	/*
@@ -677,18 +688,16 @@
 	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x80 - 0x100 */
 	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE);
 
-	/* Unprotect SCRATCHPADs: offsets: 0x100 - 0x200 */
-	for (i = 0; i < 2; i++) {
-		prot_base += SNGL_REG_PROT_SIZE;
-		regmap_writel(~0, PRIV_PROT_REG(prot_base));
-	}
-
-	/* offsets: 0x200 - 0x300 */
+	/* Unprotect SCRATCHPADs: offsets: 0x100 - 0x180 */
 	prot_base += SNGL_REG_PROT_SIZE;
-	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 2);
+	regmap_writel(~0, PRIV_PROT_REG(prot_base));
+
+	/* offsets: 0x180 - 0x300 */
+	prot_base += SNGL_REG_PROT_SIZE;
+	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 3);
 
 	/* offsets: 0x300 - 0x380 */
-	prot_base += SNGL_REG_PROT_SIZE * 2;
+	prot_base += SNGL_REG_PROT_SIZE * 3;
 	mask = ~(PROT_BIT(mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS) &
 		PROT_BIT(mmPSOC_GLOBAL_CONF_KMD_MSG_TO_CPU) &
 		PROT_BIT(mmPSOC_GLOBAL_CONF_TRACE_ADDR) &
